core/src/Kokkos_NumericTraits.hpp

```diff
@@ -135,20 +135,32 @@ template <> struct denorm_min_helper<float> { static constexpr float value = __F
 template <> struct denorm_min_helper<double> { static constexpr double value = __DBL_DENORM_MIN__; };
 template <> struct denorm_min_helper<long double> { static constexpr long double value = __LDBL_DENORM_MIN__; };
 #endif
+// GCC <10.3 is not able to evaluate T(1) / norm_min<T>::value at compile time when passing -frounding-math
+// https://godbolt.org/z/zj9svb1T7
+// Similar issue was reported on IBM Power without the compiler option
+#define KOKKOS_IMPL_WORKAROUND_CONSTANT_EXPRESSION_COMPILER_BUG
+#ifndef KOKKOS_IMPL_WORKAROUND_CONSTANT_EXPRESSION_COMPILER_BUG
 // NOTE see ?lamch routine from LAPACK that determines machine parameters for floating-point arithmetic
 template <class T>
 constexpr T safe_minimum(T /*ignored*/) {
-  constexpr auto one   = static_cast<T>(1);
-  constexpr auto eps   = epsilon_helper<T>::value;
-  constexpr auto tiny  = norm_min_helper<T>::value;
-  constexpr auto huge  = finite_max_helper<T>::value;
-  constexpr auto small = one / huge;
+  constexpr auto one  = static_cast<T>(1);
+  constexpr auto eps  = epsilon_helper<T>::value;
+  constexpr auto tiny = norm_min_helper<T>::value;
+  constexpr auto huge = finite_max_helper<T>::value;
+  constexpr auto small = one / huge;  // error: is not a constant expression
```

Comment:
Alas, C++ doesn't have conditional `constexpr` yet.

Reply:
No and that would not help if it did